showcase.yaml

```diff
@@ -1,219 +1,214 @@
-            a_install:
-              command: !Sub './install.sh -a "${AssumeRole}"'
-              cwd: '/opt'
     Properties:
       ImageId: !FindInMap [!FindInMap [OSMap, !Ref OS, RegionMap], !Ref 'AWS::Region', AMI]
       IamInstanceProfile: !Ref InstanceProfile
-      InstanceType: 't2.micro'
+      InstanceType: t2.micro
       KeyName: !If [HasKeyName, !Ref KeyName, !Ref 'AWS::NoValue']
-      UserData: !If
-      - UseUbuntu
-      - 'Fn::Base64': !Sub |
-          #!/bin/bash -x
-          bash -ex << "TRY"
-            apt-get update
-            apt-get -y install python-setuptools
-            mkdir aws-cfn-bootstrap-latest
-            curl -s -m 60 https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz | tar xz -C aws-cfn-bootstrap-latest --strip-components 1
-            easy_install aws-cfn-bootstrap-latest
-            /usr/local/bin/cfn-init --verbose --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
-          TRY
-          /usr/local/bin/cfn-signal --exit-code $? --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
-      - 'Fn::Base64': !Sub |
-          #!/bin/bash -x
-          /opt/aws/bin/cfn-init --verbose --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
-          /opt/aws/bin/cfn-signal --exit-code $? --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
+      UserData:
+        Fn::Base64: !Sub |
+          #cloud-config
+
+          packages:
+            # Install packages needed to install the cloudformation bootstrap from source
+            # (doing this to be more cross platform compatible)
+            - python-setuptools
+            - curl
+            - tar
+
+          runcmd:
+            # Install cfn tools
+            - easy_install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz
+
+            # Run cfn-init which will execute the metadata
+            - /usr/local/bin/cfn-init --verbose --stack="${AWS::StackName}" --region="${AWS::Region}" --resource=Instance
+
+            # Signal success/failure to the cloudformation stack
+            - /usr/local/bin/cfn-signal --exit-code $? --stack="${AWS::StackName}" --region="${AWS::Region}" --resource=Instance
+
       NetworkInterfaces:
       - AssociatePublicIpAddress: true
         DeleteOnTermination: true
         SubnetId: !Ref Subnet
         DeviceIndex: 0
         GroupSet:
         - !Ref SecurityGroup
+
       Tags:
       - Key: Name
-        Value: 'AWS EC2 SSH access with IAM showcase'
+        Value: AWS EC2 SSH access with IAM showcase
+
+    Metadata:
+      AWS::CloudFormation::Init:
+        config:
+          files:
+            # Pull down the install script directly from gihub
+            /opt/install.sh:
+              mode: '000755'
+              owner: root
+              group: root
+              source: !Sub https://raw.githubusercontent.com/ebmeierj/aws-ec2-ssh/${EcsSshVersion}/install.sh
```

Comment:
This should probably point to the widdix version ;-)

Reply:
I would like it to ;)